### ISSUE TITLE
API key not mine/ tumblr rate limit exceeded

### ISSUE THREAD

**Issue by draconixx** [RipMeApp/ripme#1086]
* Ripme version: 1.7.70
* Java version: Version 8 update, version 191, build (build 1.8.0_191-b12)
 Operating system: Windows 10

* Exact URL you were trying to rip when the problem occurred:  http://razsnowpaw.tumblr.com
* Please include any additional information about how to reproduce the problem: I rip that exact link, and without it even downloading anything, it just says "tumblr rate limit exceeded." It also lists an api key that isn't mine, which i have changed in the rip.properties file, but it still shows one that isn't mine.

## Expected Behavior
It should display my api key and rip the content
Detail the expected behavior here.

## Actual Behavior
Its telling me the rate limit is exceeded and displaying an api key that isn't my own,
Detail the actual (incorrect) behavior here. You can post log snippets or attach log files to your issue report.
![screenshot_6](https://user-images.githubusercontent.com/45579985/49411944-98899380-f71f-11e8-9a6e-c5e70b3c3262.png)

**Comment by GuidedByCthulhu** [comment]
Did you update the config under AppData to use your Tumblr API Key?
If not, thats your issue. 
Click on the Wiki for more info.

**Comment by draconixx** [comment]
I went to the rip.properties and updated that, but it doesn't work. By the config do you mean the properties? If not, could you elaborate?

**Comment by GuidedByCthulhu** [comment]
Yes, the rip.properties.
Are you using the most recent release?  
Check that you updated the tumblr api key, and not the twitter api key.
Make sure you saved the file, and restarted the app.

**Comment by draconixx** [comment]
Yes, I've done all of that and it still doesn't work.

**Comment by cyian-1756** [comment]
You should close ripme before editing the file

**Comment by draconixx** [comment]
It's still not working

**Comment by cyian-1756** [comment]
@draconixx 

Where is the file that you're editing?

**Comment by CarliWilliams** [comment]
The same issue is happening to me. 
I edit the line 

`tumblr.auth = JFNLu3CbINQjRdUvZibXW9VpSEVYYtiPJ86o8YmvgLZIoKyuNX`

within the file

`~/Library/Application\ Support/ripme/rip.properties `

so that tumblr.auth is set equal to my api authorization. 
When I open the jar file and run the program, it tells me file limit exceeded and further more when I check `~/Library/Application\ Support/ripme/rip.properties ` again, it shows that tumblr.auth has been set equal to JFNLu3CbINQjRdUvZibXW9VpSEVYYtiPJ86o8YmvgLZIoKyuNX again. 

To summarize, I keep changing tumblr.auth to my api key, but rip.properties reverts to tis old version once the jar file is opened.

**Comment by cyian-1756** [comment]
> To summarize, I keep changing tumblr.auth to my api key, but rip.properties reverts to tis old version once the jar file is opened.

@CarliWilliams Be sure to close ripme before editing the config file

**Comment by CarliWilliams** [comment]
@cyian-1756 I do close it. When I open it again, the api key changes back. I think this line in TumblrRipper.jave is the problem `String userDefinedAPIKey = Utils.getConfigString(TUMBLR_AUTH_CONFIG_KEY, "JFNLu3CbINQjRdUvZibXW9VpSEVYYtiPJ86o8YmvgLZIoKyuNX");`

**Comment by cyian-1756** [comment]
> I think this line in TumblrRipper.java is the problem String userDefinedAPIKey = Utils.getConfigString(TUMBLR_AUTH_CONFIG_KEY, "JFNLu3CbINQjRdUvZibXW9VpSEVYYtiPJ86o8YmvgLZIoKyuNX");

No that line is unrelated

**Comment by nyrizo** [comment]
> @cyian-1756
> 
> The same issue is happening to me.
> I edit the line
> 
> `tumblr.auth = JFNLu3CbINQjRdUvZibXW9VpSEVYYtiPJ86o8YmvgLZIoKyuNX`
> 
> within the file
> 
> `~/Library/Application\ Support/ripme/rip.properties `
> 
> so that tumblr.auth is set equal to my api authorization.
> When I open the jar file and run the program, it tells me file limit exceeded and further more when I check `~/Library/Application\ Support/ripme/rip.properties ` again, it shows that tumblr.auth has been set equal to JFNLu3CbINQjRdUvZibXW9VpSEVYYtiPJ86o8YmvgLZIoKyuNX again.
> 
> To summarize, I keep changing tumblr.auth to my api key, but rip.properties reverts to tis old version once the jar file is opened.

I'm having the same exact problem.  I am changing the API key, to my own key, in the rip.properties file but it's using a different one.  Thanks for any help you can offer!

**Comment by rautamiekka** [comment]
Remove the write permission from the file, that should prevent it from be changed.

**Comment by nyrizo** [comment]
> Remove the write permission from the file, that should prevent it from be changed.

I have done that. But anyway, the file does not change the API key back to default.  My API key is listed correctly in the rip.properties and does not change.  However, it's using a different API key (perhaps the default one that came with the app?).: api_key=qpdkY6nMknksfvYAhf2xIHp0iNRLkMlcWShxqzXyFJRxIsZ1Zz&offset=0  Where is it getting that key from?  Thank you!

**Comment by Hrxn** [comment]
> api_key=qpdkY6nMknksfvYAhf2xIHp0iNRLkMlcWShxqzXyFJRxIsZ1Zz

But that's not the same as mentioned earlier in the thread? So is it working for you?

**Comment by nyrizo** [comment]
> > api_key=qpdkY6nMknksfvYAhf2xIHp0iNRLkMlcWShxqzXyFJRxIsZ1Zz
> 
> But that's not the same as mentioned earlier in the thread? So is it working for you?

Unfortunately, it's not working.  I am not the same person who posted earlier in the thread. I'm having the same problem, and it is not resolved.  Thanks for any help you can offer!

**Comment by cyian-1756** [comment]
@nyrizo 

@CarliWilliams 

Please try the following

1) Reboot your cmputer

2) Ensure that ripme is not running, kill it if it is

3) Edit the file at 

%LOCALAPPDATA%\ripme on Windows

$HOME/.config/ripme on *Nix

$HOME/Library/Application Support/ripme on MacOS

Abd replace the like "tumblr.auth = JFNLu3CbINQjRdUvZibXW9VpSEVYYtiPJ86o8YmvgLZIoKyuNX" with "tumblr.auth = YOUR KEY GOES HERE"

**Comment by nyrizo** [comment]
> 1. Reboot your cmputer
> 2. Ensure that ripme is not running, kill it if it is
> 3. Edit the file at
> $HOME/Library/Application Support/ripme on MacOS
> 
> Abd replace the like "tumblr.auth = JFNLu3CbINQjRdUvZibXW9VpSEVYYtiPJ86o8YmvgLZIoKyuNX" with "tumblr.auth = YOUR KEY GOES HERE"

Hi @cyian-1756, thank you so much.  I restarted my computer, made sure RipMe wasn't running (I even checked the activity monitor to double check it wasn't running).  I edited the file (though it already my api key).  I ran RipMe and I got the same error :(  The one thing that I can think of, now that I'm thinking back, is that the config file wasn't automatically installed (I downloaded it off this website and installed it in the right place).  However, it *was* working but then stopped so I'm sure I put the config file in the right place...  I'm not sure what else to try!  Thank you again!

**Comment by rautamiekka** [comment]
RipMe 1.7.72 does work with Tumblr when you give the correct key, specifically the consumer one; I recently used one to rip my Tumblr likes and it was a blast. You probly wanna use 2 threads instead of 1, but I wouldn't recommend using more than that so they don't auto-ban or throttle or something.

**Comment by cyian-1756** [comment]
@nyrizo 

Try closing ripme and copying the config file that had your api key to the same dir as ripme.jar and then start ripme/ This should make ripme use that config

**Comment by nyrizo** [comment]
> @nyrizo
> 
> Try closing ripme and copying the config file that had your api key to the same dir as ripme.jar and then start ripme/ This should make ripme use that config

Thank you, @cyian-1756 moving the config file worked!  Thanks for taking time to help me out!

**Comment by cyian-1756** [comment]
@nyrizo 

No problem!

@draconixx @CarliWilliams 

You might also want to try editing and moving the config file

**Comment by harl** [comment]
Where would I find my own tumblr API/auth keys whatever?
Surely I don't have to manually go through the that OAuth registration process for that?

**Comment by rautamiekka** [comment]
> Where would I find my own tumblr API/auth keys whatever?
> Surely I don't have to manually go through the that OAuth registration process for that?

Once you've regged a key -and- it's still active, you'll find them listed and perfectly accessible at https://www.tumblr.com/settings/apps (fastest to reach) and https://www.tumblr.com/oauth/apps

**Comment by harl** [comment]
Thanks.. So, I *do* have to manually register this as an "app" specifically for my own account.
Well then.
